### ISSUE TITLE
Dependency system for Mineunit modules

### ISSUE THREAD

**Issue by S-S-X** [S-S-X/mineunit#38]
As Mineunit is kind of built from modules and those are growing pretty fast it would be good to have well defined dependencies between modules.
Currently dependencies are handled by just implicitly loading dependencies when module is included in test set.

For example `server` module requires few others:
https://github.com/S-S-X/mineunit/blob/3ab99b0f866917ac392829cb5cd61c8742d30511/server.lua#L5-L9

And `itemstack` requires `metadata`:
https://github.com/S-S-X/mineunit/blob/3ab99b0f866917ac392829cb5cd61c8742d30511/itemstack.lua#L1-L3

It would be good to have all these in one place so that loading order could be handled better, currently there's no well defined module loading order especially if modules are included individually.

However there's some upsides with not having well defined initialization order: it enforces writing code with better fault tolerance.
Basically it is harder to make shortcuts, global variables cannot be trusted.

It would be good if dependency system wont take away those restrictions but to enforce that quality it probably requires some way to test modules while ignoring dependencies.

**Comment by S-S-X** [comment]
One example why current system is not nice, following will work fine:
```lua
require("mineunit")
mineunit("player")
mineunit("protection")
mineunit("common/after")
```
But this will fail with error not related to protection module in any way:
```lua
require("mineunit")
mineunit("player")
--mineunit("protection")
mineunit("common/after")
```
In this case actual error is missing `core.register_globalstep`
```
0 successes / 0 failures / 1 error / 0 pending : 0.00436 seconds

Error → .../mineunit/common/after.lua @ 5
suite spec/init_spec.lua
.../mineunit/common/after.lua:5: attempt to call field 'register_globalstep' (a nil value)
```
